nucypher/blockchain/eth/trackers/dkg.py

```diff
@@ -68,6 +70,9 @@ class ActiveRitualTracker:
 
     MAX_CHUNK_SIZE = 10000
 
+    # how often to check/purge for expired cached values - 8hrs?
```

Comment:
👍 The tricky is part is are we happy with this way of purging these items. In this case we're piggy-backing on ritual events instead of having a dedicated task. Using a dedicated periodic running task just felt excessive.

The one edge case for this current logic is if a bunch of rituals get created initially (like hundreds etc.) before the 8-hour timeline, and then no more rituals afterwards. Without subsequent events the cache doesn't clear, unless of course the node restarts. Not the end of the world in my mind, it's 3 booleans stored per ritual which is pretty small.

Reply:
Good reasoning. 8 hrs looks fine for this edge case. And as you say, the size of the stored data is pretty small, I don't think we will have problems even with larger periods of time.

Reply:
Considering this cache only grows in size with new DKG activities, piggy-backing is still an effective state pruning strategy here 👍🏻

Reply:
👍 - great point.